File: backend/src/api/users/Users_Model.ts
import { Entity, Column, PrimaryColumn } from 'typeorm';
import { generateUUID } from '../../utils/uuid';

@Entity('users')
export class Users {
    @PrimaryColumn('uuid')
    id: string;

    @Column()
    name: string;

    @Column()
    email: string;

    @Column()
    password: string;

    constructor() {
        if (!this.id) {
            this.id = generateUUID();
        }
    }
}

File: app/lib/src/data/Users/users_model.dart
class Users {
  final String? id;
  final String name;
  final String email;
  final String password;

  Users({
    this.id,
    required this.name,
    required this.email,
    required this.password,
  });

  factory Users.fromMap(Map<String, dynamic> map) {
    return Users(
      id: map['id'],
      name: map['name'],
      email: map['email'],
      password: map['password'],
    );
  }
}
